app/views/contexts/summary/group.py

```diff
@@ -77,10 +91,46 @@ def _build_blocks(
                     ]
                 )
 
-        return blocks
+            elif block["type"] in ["ListCollector"]:
+                list_collector_block = ListCollectorBlock(
+                    routing_path=routing_path,
+                    answer_store=answer_store,
+                    list_store=list_store,
+                    progress_store=progress_store,
+                    metadata=metadata,
+                    response_metadata=response_metadata,
+                    schema=schema,
+                    location=location,
+                    language=language,
+                )
+
+                section = schema.get_section(location.section_id)
+                summary = section.get("summary", {})
+                if summary_items := summary.get("items"):
+                    for summary_item in summary_items:
```

Comment:
This shouldn't loop through all summary items, it should get the summary items that matches the for_list of the list collector.

Reply:
Added the `for_list` check now, thanks!